src/FurCoNZ.Web/Controllers/AccountController.cs

```diff
@@ -33,33 +33,35 @@ public AccountController(IOrderService orderService, IUserService userService)
 
         public async Task<IActionResult> Index(CancellationToken cancellationToken = default)
         {
-            return View(new AccountViewModel(await _userService.GetCurrentUserAsync(cancellationToken)));
+            var hasRedirect = HttpContext.Session.Keys.Contains(RedirectAfterDetailsSessionKey);
+            var user = await _userService.GetCurrentUserAsync(cancellationToken);
+            return View(new AccountViewModel(user)
+                {
+                    IsConfirmingDetails = hasRedirect,
+                });
         }
 
-        [HttpGet, HttpPost]
+        [HttpPost]
+        [ValidateAntiForgeryToken]
```

Comment:
Whoops, it turns out this isn't on by default. Good call here, but I'll fix this up across the project in #55.